neurom/stats.py

```diff
@@ -132,3 +132,32 @@ def scalar_stats(data, functions=('min', 'max', 'mean', 'std')):
         stats[func] = getattr(np, func)(data)
 
     return stats
+
+
+def compare_two(data1, data2, test='ks_2samp'):
+    '''Compares two distributions of data
+       and assess two scores: a distance between them
+       and a probability they are drawn from the same
+       distribution.
+
+    Parameters:
+        data1: numpy array of dataset 1
+        data2: numpy array of dataset 2
+        test: string\
+            Defines the statistical test to be used, based\
+            on the scipy available modules.\
+            Accepted tests: ks_2samp(default), mannwhitneyu
+
+    Returns:
+        dist: float\
+            High numbers define high dissimilarity between the two datasets
+        p-value: float\
+            Small numbers define high probability the data come from\
+            same dataset.
+    '''
+    results = getattr(_st, test)(data1, data2)
+
+    stats = {'dist': results[0]}
+    stats['p-value'] = results[1]
```

Comment:
Hi,

first: I would take all ttests of scipy.stats in neuroM (middleterm), depending on how much work it is to integrate them.

second: which default? Depends on which is the most frequent case the customers have. I would take either the  scipy.stats.ttest_ind(a, b, axis=0, equal_var=True)  if the the distributions are more or less unimodal and symmetric or the mannwhitneyu otherwise (if they are highly skewed and obviously non-normal, for example). All the above tests work for continuous distributions ONLY. Special attention should be paid to equal_var=True or False depending on what one has. It switches between two different tests.

The reason not to make the kstest the default is that it is sensitive to all kinds of shape differences of the distributions. The tests mentioned above focus on differences in location parameters like the means (ttest) or medians (mannwhittneyu).

Another caution point is the wether the two samples are correlated (“related" in the scipy.stats language) or not. The corresponding versions of the tests are: scipy.stats.ttest_rel(a, b, axis=0)  for the ttest_ind and for the mannwhitneyu is the wilcoxon.

ATTENTION!!! the following statement is wrong. It’s the other way around: small p-values point to significantly different distributions. More specific: if pval < 0.05 then we reject the null hypothesis that the two distributions are the same. If pval>=0.05 we cannot reject the null, which DOES NOT MEAN THAT THE TWO DATA SETS COME FROM THE SAME DISTRIBUTION. What I say in this case is that: there is no evidence for significant difference, thus I do not undertake any action (to modify the system, for example).

I hope that helps, have fun,

nancy



> +        p-value: float\
> +            Small numbers define high probability the data come from\

> + same dataset.
<http://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ttest_ind.html#scipy.stats.ttest_ind>
On 19 Jan 2016, at 18:23, lidakanari <notifications@github.com<mailto:notifications@github.com>> wrote:


In neurom/stats.py<https://github.com/BlueBrain/NeuroM/pull/220#discussion_r50143496>:

> +        test: string\
> +            Defines the statistical test to be used, based\
> +            on the scipy available modules.\
> +            Accepted tests: ks_2samp(default), mannwhitneyu
> +
> +    Returns:
> +        dist: float\
> +            High numbers define high dissimilarity between the two datasets
> +        p-value: float\
> +            Small numbers define high probability the data come from\
> +            same dataset.
> +    '''
> +    results = getattr(_st, test)(data1, data2)
> +
> +    stats = {'dist': results[0]}
> +    stats['p-value'] = results[1]


A named tuple is a good idea. Will change to that!

—
Reply to this email directly or view it on GitHub<https://github.com/BlueBrain/NeuroM/pull/220/files#r50143496>.